benches/bigint.rs

```diff
@@ -2,10 +2,33 @@
 
 extern crate test;
 extern crate num;
+extern crate rand;
 
 use std::mem::replace;
 use test::Bencher;
 use num::{BigUint, Zero, One, FromPrimitive};
+use num::bigint::RandBigInt;
+use rand::{SeedableRng, StdRng};
+
+fn multiply_bench(b: &mut Bencher, xbits: usize, ybits: usize) {
+    let seed: &[_] = &[1, 2, 3, 4];
+    let mut rng: StdRng = SeedableRng::from_seed(seed);
+
+    let x = rng.gen_bigint(xbits);
+    let y = rng.gen_bigint(ybits);
+
+    b.iter(|| &x * &y);
+}
+
+fn divide_bench(b: &mut Bencher, xbits: usize, ybits: usize) {
+    let seed: &[_] = &[1, 2, 3, 4];
+    let mut rng: StdRng = SeedableRng::from_seed(seed);
+
+    let x = rng.gen_bigint(xbits);
+    let y = rng.gen_bigint(ybits);
+
+    b.iter(|| &x / &y);
+}
```

Comment:
On Mon, Nov 30, 2015 at 03:38:38PM -0800, Josh Stone wrote:
> > +
> > +    let x = rng.gen_bigint(xbits);
> > +    let y = rng.gen_bigint(ybits);
> > +
> > +    b.iter(|| &x * &y);
> > +}
> > +
> > +fn divide_bench(b: &mut Bencher, xbits: usize, ybits: usize) {
> > +    let seed: &[_] = &[1, 2, 3, 4];
> > +    let mut rng: StdRng = SeedableRng::from_seed(seed);
> > +
> > +    let x = rng.gen_bigint(xbits);
> > +    let y = rng.gen_bigint(ybits);
> > +
> > +    b.iter(|| &x / &y);
> > +}
> 
> How about some pre-iter assertions that these huge divisions are actually correct?  e.g. check the round trip to multiply and add the remainder back.

Well, then it wouln't be just benchmarking division - I do have a test that does
that though, I didn't push it because it takes way too long to run if you don't
do a release build.

I don't know a good way of putting in these kinds of torture tests where you
want it to run "as long as you have time for", not any fixed amount of time. I
guess I could just make it a #[ignore] test, like you suggested for the other
division test.